src/agent/src/workload/runner.rs

```diff
@@ -1,26 +1,28 @@
-use super::config::{Action, Config};
 use crate::{
-    agents::{rust, Agent, Language},
+    agents::{rust, Agent, AgentOutput, Language},
+    workload::config::Action,
     AgentResult,
 };
 
+use super::config::Config;
+
 /// Runner for a workload.  
 /// Will execute the workload based on the inner agent (language).
 pub struct Runner {
     config: Config,
-    agent: Box<dyn Agent>,
+    agent: Box<dyn Agent + Sync + Send>,
 }
 
 impl Runner {
     pub fn new(config: Config) -> Self {
-        let agent: Box<dyn Agent> = match config.language {
+        let agent: Box<dyn Agent + Sync + Send> = match config.language {
             Language::Rust => Box::new(rust::RustAgent::from(config.clone())),
         };
 
         Runner { config, agent }
     }
 
-    pub fn run(&self) -> AgentResult<()> {
+    pub fn run(&self) -> AgentResult<AgentOutput> {
```

Comment:
You only return the logs of the `prepare`, `run` and run logs of `prepare-and-run`?

Reply:
we could do it later, it works like this

Reply:
The goal is to stream it.